src/server/game/Quests/QuestDef.cpp

```diff
@@ -227,14 +227,17 @@ uint32 Quest::XPValue(Player* player) const
     return 0;
 }
 
-int32 Quest::GetRewOrReqMoney() const
+int32 Quest::GetRewOrReqMoney(Player const* player) const
 {
     // RequiredMoney: the amount is the negative copper sum.
-    if (_rewardMoney <= 0)
+    if (_rewardMoney < 0)
         return _rewardMoney;
 
     // RewardMoney: the positive amount
-    return int32(_rewardMoney * sWorld->getRate(RATE_MONEY_QUEST));
+    if (!player || player->GetLevel() < sWorld->getIntConfig(CONFIG_MAX_PLAYER_LEVEL) || HasFlag(QUEST_FLAGS_NO_MONEY_FROM_XP))
+        return int32(_rewardMoney * sWorld->getRate(RATE_MONEY_QUEST));
+    else
+        return GetXPReward(player) * 6 * sWorld->getRate(RATE_MONEY_MAX_LEVEL_QUEST); // At level cap, any potential experience rewarded by the quest is converted to money at a rate of 6 coppers per 1 point of exp
```

Comment:
should this not be added to the reward value listed above?

Reply:
Nope, the 6 * exp is the total reward, not the bonus added to the base money reward.

Reply:
are there any quests which have a very high "regular" money reward? should it be the maximum of the two?

Reply:
I just found a similar case. The [Amphitheather of Anguish](https://www.wowhead.com/quest=12932) quests give 32100 exp, that converts into 19g 26s 00c. The quest itself gives 19g 50s 00c, and from [this video](https://youtu.be/JZb2JF9SNHA?t=10) it seems that the player at level 80 gets 19g 50s.

Reply:
So it should be `std::max` of the two then?

Reply:
Yeah, it makes sense. I'm searching for more similar cases but that seems the way it works, because it doesn't make sense that you can get less money for the same quest completed at a higher level.